src/handlers/messages.ts

```diff
@@ -204,6 +225,41 @@ export async function createMessage(
   ).data.event_id
 }
 
+/**
+ * Send a File to Synapse
+ * @param user_id The user the media will be posted by
+ * @param ts The timestamp to which the file will be dated
+ * @param filePath the path on the local filesystem
+ * @param fileName the filename
+ * @param content_type: Content type of the file
+ * @returns The Matrix Message/event ID
+ */
+export async function uploadFile(
+  user_id: string,
+  ts: number,
+  filePath: string,
+  fileName: string,
+  content_type: string
+): Promise<string> {
+  const fileStream = fs.createReadStream(filePath)
+  const accessToken = await getAccessToken(user_id)
+  log.http(`Uploading ${fileName}...`)
+
+  return (
+    await axios.post(
+      `/_matrix/media/v3/upload?user_id=${user_id}&ts=${ts}&filename=${fileName}`,
+      fileStream,
+      {
+        headers: {
+          'Content-Type': content_type,
+          'Content-Length': fs.statSync(filePath).size,
```

Comment:
Why not `(await fs.stat(filePath)).size`?

Reply:
Done with the error handling for https://github.com/verdigado/rocketchat2matrix/pull/32#discussion_r1695066932